lib/graphql/query/serial_execution/field_resolution.rb

```diff
@@ -16,14 +16,14 @@ def initialize(ast_node, parent_type, target, query, execution_strategy)
 
         def result
           result_name = ast_node.alias || ast_node.name
-          result_value = begin
-            get_finished_value(get_raw_value)
+          raw_value = begin
+            get_raw_value
           rescue GraphQL::ExecutionError => err
             err.ast_node = ast_node
             query.context.errors << err
-            nil
+            err
           end
-          { result_name => result_value  }
+          { result_name => get_finished_value(raw_value) }
```

Comment:
Good point.

Thinking through this more, I'm not even sure what we'd do if we rescued an error from one of those sources. We can't return `nil` because then we're breaking spec if the field is non-null, unless we tried to recursively call `get_finished_value` which would make it tricky to avoid infinite loops?

I'm also not sure when one of those methods would want to intentionally raise an `ExecutionError`... most such errors I expect would be of the wholly-unintentional type raised by ruby directly.